OsmAnd/src/net/osmand/plus/rastermaps/OsmandRasterMapsPlugin.java

```diff
@@ -582,10 +587,9 @@ public void toggleUnderlayState(@NonNull MapActivity mapActivity,
 			exMapTypePreference = settings.MAP_UNDERLAY_PREVIOUS;
 			map = underlayLayer.getMap();
 		}
-
-		boolean isChecked = map == null;
+		boolean isChecked = map != null;
 		boolean showSeekbar = isChecked && RasterMapMenu.isSeekbarVisible(app, type);
-		boolean hideSeekbar = !isChecked && RasterMapMenu.isSeekbarVisible(app, type);
+		boolean hideSeekbar = !isChecked;
 		MapActivityLayers mapLayers = mapActivity.getMapLayers();
 		CommonPreference<LayerTransparencySeekbarMode> seekbarModePref = settings.LAYER_TRANSPARENCY_SEEKBAR_MODE;
 		if (showSeekbar) {
```

Comment:
Did you trace showTransparencyBar, hideTransparencyBar, setTransparencyBarEnabled methods across the code? There might be more bugs related to show/hide the bar.